archaius2-core/src/main/java/com/netflix/archaius/config/AbstractConfig.java

```diff
@@ -298,6 +299,29 @@ protected <T> T getValue(Type type, String key) {
     @SuppressWarnings("unchecked")
     protected <T> T getValueWithDefault(Type type, String key, T defaultValue) {
         Object rawProp = getRawProperty(key);
+        if (rawProp == null && type instanceof ArchaiusType) {
```

Comment:
I don't ... hate this, but let's think about it a bit more. I'd really love it if there was some way we could call the existing decoders here instead of marshaling the entire map into a single string just to unmarshal it back a nanosecond later. The same goes for the List support.

Reply:
I updated the code to instead rely on the decoder inside the AbstractConfig. The typing does become a little more annoying, and I also didn't include some of the DefaultCollectionsTypeConverterFactory types (specifically SortedSet and LinkedList). Honestly no great reason for that, but our SBN docs only specifically reference Map, List, and Set.